File: training-data/java/FloodFill.java
import java.util.ArrayDeque;
import java.util.Deque;

public class FloodFill {
    public static int fill(char[][] grid, int sr, int sc, char color) {
        char target = grid[sr][sc];
        if (target == color) return 0;
        Deque<int[]> queue = new ArrayDeque<>();
        queue.add(new int[] {sr, sc});
        grid[sr][sc] = color;
        int count = 1;
        int[][] dirs = {{1, 0}, {-1, 0}, {0, 1}, {0, -1}};
        while (!queue.isEmpty()) {
            int[] p = queue.poll();
            for (int[] d : dirs) {
                int r = p[0] + d[0], c = p[1] + d[1];
                if (r >= 0 && r < grid.length && c >= 0 && c < grid[0].length && grid[r][c] == target) {
                    grid[r][c] = color;
                    count++;
                    queue.add(new int[] {r, c});
                }
            }
        }
        return count;
    }

    public static void main(String[] args) {
        char[][] img = {"..##.".toCharArray(), ".#..#".toCharArray(), ".#..#".toCharArray(), "..##.".toCharArray()};
        System.out.println("filled " + fill(img, 1, 2, 'o'));
        for (char[] row : img) System.out.println(new String(row));
    }
}

File: training-data/go/flood_fill.go
package main

import "fmt"

func floodFill(img [][]byte, r, c int, color byte) int {
	target := img[r][c]
	if target == color {
		return 0
	}
	var fill func(r, c int) int
	fill = func(r, c int) int {
		if r < 0 || c < 0 || r >= len(img) || c >= len(img[r]) || img[r][c] != target {
			return 0
		}
		img[r][c] = color
		return 1 + fill(r+1, c) + fill(r-1, c) + fill(r, c+1) + fill(r, c-1)
	}
	return fill(r, c)
}

func main() {
	img := [][]byte{[]byte("..####.."), []byte(".#....#."), []byte(".#....#."), []byte("..####..")}
	fmt.Println("filled", floodFill(img, 1, 3, 'o'))
	for _, row := range img {
		fmt.Println(string(row))
	}
}
